### PR TITLE
cassandra pool verify partitioner on node entrance

### PR DESCRIPTION
**Pull request by clockfort**

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java [hunk 1]
```diff
@@ -515,4 +497,20 @@ static boolean isRetriableException(Throwable t) {
...
+        }
+    };
+
+    final FunctionCheckedException<Cassandra.Client, Void, Exception> createInternalMetadataTable = new FunctionCheckedException<Cassandra.Client, Void, Exception>() {
+        @Override
+        public Void apply(Cassandra.Client client) throws Exception {
+            createTableInternal(client, CassandraConstants.METADATA_TABLE);
+            return null;
+        }
+    };
```
Comment (schlosna) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java]: Should these move to CassandraVerifier?

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java [hunk 2]
```diff
@@ -207,7 +205,7 @@ private boolean isHostHealthy(InetSocketAddress host) {
         try {
             CassandraClientPoolingContainer testingContainer = new CassandraClientPoolingContainer(host, config);
             testingContainer.runWithPooledResource(describeRing);
-            testingContainer.runWithPooledResource(CassandraVerifier.healthCheck);
+            testingContainer.runWithPooledResource(validatePartitioner);
```
Comment (schlosna) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java]: Is CassandraVerifier.healthCheck still used (actually seems to describe_version)?

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java (hunk 1 above)
Comment (clockfort) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java]: not trivially, they need non-static Cassandra config and it's not easy to pass in both to a FunctionCheckedException

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java (hunk 2 above)
Comment (clockfort) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java]: healthCheck is very very basic and can only fail if the node is truely buggered.

validatePartitioner is strictly speaking a superset of functionality testing.

Comment (clockfort) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java]: and yes, healthCheck is still used, it has its place in the code I wrote in the last PR to determine if hosts are so badly screwy during the bootup process that we shouldn't even try doing anything else with them.

**Pull request merged**